woocommerce/woocommerce-ios
# [Mobile Payments] Display last4 and card brand on refund screen consistently

Closes: #5976 


### Description

Previously, the refund screen was not displaying the card details consistently, especially missing them for orders where the payment had just been taken.

When loading the order again, the `update(with:)` function was not setting the chargeID we'd recieved from the network on the storage, it was only set when the order was first inserted from a new fetch.

Setting `chargeID` in `update(with:)` fixed these issues, however, @koke's suggestion that they may be caused by the list fetch overwriting the `chargeID` with nil because of the fields on the list fetch not including `meta_data` needed investigation.

I found that they were indeed being overwritten, but then we would fetch fresh data again including the chargeID when entering the Order Details screen before the refund, so in general the chargeID was always present.

To reduce the risk of a race condition or other future bugs here, I changed all Order requests to fetch `meta_data`. With this, I can see that `chargeID` doesn't get incorrectly set to `nil` any more. The `singleOrderExtraFields` was initially added to avoid fetching `line_items` for every order when viewing the list, for performance reasons. We will discuss whether or not this is desirable, and whether we should provide for this in the storage layer with flags to say "don't update these fields".

### Testing instructions

To test, a WooCommerce Payments In Person Payment setup is required. See PdfdoF-D-p2 for info.

1. In the Orders tab, tap an order which has been paid for by a card-present payment
2. Tap `Refund`
3. Select a product to refund or toggle `Refund fees` if required, then tap `Next`
4. Observe that the `Refund Via` row contains card brand and last 4 digits

You can use the [testing card numbers that Stripe provide](https://stripe.com/docs/terminal/references/testing#standard-test-cards) to test different brands/last4 combos with the simulated reader.

### Screenshots

![Updated Mastercard](https://user-images.githubusercontent.com/2472348/153887329-b1dcb5ac-cfe7-47e6-b76e-50f3a3181a25.jpg)


---
